sda/cmd/auth/main.go

```diff
@@ -422,6 +438,8 @@ func main() {
 	// Endpoint for client login info
 	app.Get("/info", authHandler.getInfo)
 
+	defer authHandler.Config.DB.Close() // needs to be after Fatalf
```

Comment:
The problem is not that this was before a `log.Fatalf` earlier, where it should be, but rather the existence of the `log.Fatal` on line 435.

There are a couple of different ways to handle this:
1. Place the defer in line 415 **AND** change the `log.Fatalf` to a `log.Panicf`
2. Move lines 433-436 up to line 368
3. Add a `shutdown` function like [here:](https://github.com/neicnordic/sensitive-data-archive/blob/bcfd30da7d37801e83b8cd49b61d633156cbd127/sda/cmd/api/api.go#L140-L144) and call that function on exit like [this](https://github.com/neicnordic/sensitive-data-archive/blob/bcfd30da7d37801e83b8cd49b61d633156cbd127/sda/cmd/api/api.go#L79-L82).

Reply:
Moving 433-436 up above the database connection bit shouldn't be an issue.

Anyway, this app reeds some refactoring in order to improve logic and readability later on, som we can fix all the small stuff then.